### PR TITLE
[Splitwise] New Components

### LINKED ISSUE #4149: [ACTIONS, TRIGGER] - Splitwise new components
**App**
Splitwise
https://dev.splitwise.com/

**Currently there is no workaround.**

**I would like to have triggers that indicate new payments are made (to/from myself), groups are made, and re-occuring notices about debts. I would also like to get the amount balance from each payment, current payments, and information about each information (amount, to-from, photo, details). As a stretch I would also like to get all of the raw payment information for a time period (i.e. to get a budget breakdown for a report or reimbursement).**

### PR DESCRIPTION
**Pull request by andrewjschuang**
**Actions:**

- Create Expense
- Get Expense Information
- Get Expenses

**Sources:**

- Expense Created
- Group Created
- Recently Occurred Activity

Resolves #4149.

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4501"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/splitwise/splitwise.app.mjs [hunk 1]
```diff
@@ -1,11 +1,145 @@
...
+    baseUrl() {
+      return "https://secure.splitwise.com/api/v3.0";
+    },
+    async _makeRequest({
+      $ = this, path, ...opts
+    }) {
+      return axios($, {
+        url: this.baseUrl() + path,
+        headers: {
+          Authorization: `Bearer ${this.$auth.oauth_access_token}`,
```
Comment (lcaresia) [inline components/splitwise/splitwise.app.mjs:72]: Move this to a getter method

```suggestion
          Authorization: `Bearer ${this._accessToken()}`,
```

**Review by lcaresia — CHANGES_REQUESTED** [review]
Thanks for your contribution!

Is needed just one fix.

**Review by lcaresia — DISMISSED** [review]
LGTM!

**Review by lcaresia — APPROVED** [review]

**Pull request merged**